### DISCUSSION TITLE
Seeking Guidance on Testing Custom Language Plugins with LSP4IJ

### DISCUSSION THREAD

**Discussion by RehanChalana** [redhat-developer/lsp4ij#548] [Q&A]
Hello Everyone,

I am working on a KCL-lang IntelliJ plugin utilizing LSP4IJ to implement Language Server Protocol (LSP) support. 

Could anyone provide guidance or best practices for writing integration tests for a plugin using LSP4IJ?

Any advice, examples, or resources would be greatly appreciated!

**Comment by angelozerr** [comment]
We have written several tests in LSP4IJ for LSP completion, rename, etc in https://github.com/redhat-developer/lsp4ij/tree/main/src/test/java/com/redhat/devtools/lsp4ij/features

But those tests uses a Mock language server. Perhaps you could adapt those classes to test your LS and better contribute to LSP4IJ to provide generic classes to test any language server?

**Reply by RehanChalana** [reply]
Sounds exactly like what I was looking for!
Perhaps you meant [main/src/test/java/com/redhat/devtools/lsp4ij/fixtures](https://github.com/redhat-developer/lsp4ij/tree/main/src/test/java/com/redhat/devtools/lsp4ij/fixtures)?

I would love to contribute as well!

However, I encountered an error when trying to import these classes. Could you help me? Does this issue have anything to do with the fact that these classes are defined in the test directory rather than main?

**Reply by angelozerr** [reply]
Perhaps but to be honnest with you I don't know. I have just written this class to test LSP4IJ and I have never provide in IJ a kind of Test framework.

Any contribution are welcome!

**Comment by angelozerr** [comment]
I close this discussion in favor of https://github.com/redhat-developer/lsp4ij/issues/554
